File: C#SrcFiles/FireMageSpawnerScript.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class FireMageSpawnerScript : MonoBehaviour
{
    public GameObject FireMage;
    private float Timer;
    private bool startTimer = false;

    // Start is called before the first frame update
    void Start()
    {
        // at start the timer is 0.
        Timer = 10;
        // adding the function to the Event!
        SlimeScript.CharachterEvent += StartEnemyTimer;
    }

    // Update is called once per frame
    void Update()
    {
        // Creating  a timer for first enemy to come;
        if (startTimer && Timer <= 0)
        {
            SendEnemy();
            startTimer = false;
        }
        else if (startTimer && Timer > 0)
        {
            Timer -= 1 * Time.deltaTime;
            print(Timer);
        }

    }

    private void StartEnemyTimer()
    {
        startTimer = true;
    }

    private void SendEnemy()
    {
        //Instantiate(FireMage, transform.position, transform.rotation);
    }

}

File: C++SrcFiles/FireMageSpawnerScript.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cmath>
#include <algorithm>

using namespace std;

class FireMageSpawnerScript : public MonoBehaviour
{
private:
    GameObject* fireMagePrefab;
    float timer;
    bool startTimer;

public:
    FireMageSpawnerScript()
    {
        timer = 10.0f;
        startTimer = false;

        // Add the callback function to the event
        SlimeScript::CharacterEvent += StartEnemyTimer;
    }

    void Update()
    {
        // Check if the timer is expired and send an enemy
        if (startTimer && timer <= 0.0f) {
            SendEnemy();
            startTimer = false;
        } else if (startTimer && timer > 0.0f) {
            timer -= 1.0f * Time.deltaTime;
            cout << timer << endl;
        }
    }

    void StartEnemyTimer()
    {
        startTimer = true;
    }

    void SendEnemy()
    {
        // Instantiate the FireMage object at the spawner's position
        Instantiate(fireMagePrefab, transform->position, transform->rotation);
    }
};
